src/tests/createHeadlessForm.test.js

```diff
@@ -1035,6 +1035,94 @@ describe('createHeadlessForm', () => {
       );
     });
 
+    it('support "date" field type with a minDate', () => {
+      const { fields, handleValidation } = createHeadlessForm(schemaInputTypeDate);
+
+      const validateForm = (vals) => friendlyError(handleValidation(vals));
+
+      expect(fields[0]).toMatchObject({
+        label: 'Birthdate',
+        name: 'birthdate',
+        required: true,
+        schema: expect.any(Object),
+        type: 'date',
+        maxLength: 10,
+        minDate: '1922-03-01',
+        maxDate: '2022-03-01',
+      });
+
+      expect(validateForm({})).toEqual({
+        birthdate: 'Required field',
+      });
+
+      const todayDateHint = new Date().toISOString().substring(0, 10);
+
+      expect(validateForm({ birthdate: '' })).toEqual({
+        birthdate: `Must be a valid date in yyyy-mm-dd format. e.g. ${todayDateHint}`,
+      });
+
+      expect(validateForm({ birthdate: '1922-02-01' })).toEqual({
+        birthdate: 'The date must be 1922-03-01 or after.',
+      });
+
+      expect(validateForm({ birthdate: '1922-03-01' })).toEqual(undefined);
+
+      expect(validateForm({ birthdate: '2021-03-01' })).toEqual(undefined);
+    });
+
+    it('support "date" field type with a maxDate', () => {
+      const { fields, handleValidation } = createHeadlessForm(schemaInputTypeDate);
```

Comment:
nit: Could you refactor the `schemaInputTypeDate` to be a plain object without `JSONSchemaBuilder`?

Also, in the `mockDateInput`, the `maxLength` is not necessary as the format is enough to validate the value properly.